### PR TITLE
Docs: Adds a landing page for reference docs

### PR DESCRIPTION
**Pull request by kiwicopple**
## What kind of change does this PR introduce?

dds a landing page for reference docs

## What is the current behavior?

Currently the landing page when you click on "Reference" goes to `/docs/reference/javascript/supabase-client`, which is a page "under" the Javascript client.

## What is the new behavior?

This change:

- Adds a new page which is more generic
- Removes the old page
- Adds a redirect to the new generic page

image

## Additional context

There is one blocker. There is a new "Overview" in the sidebar:

image

It has two issues:

- [x] the styling is off (especially the padding/alignment)
- [x] it isn't clickable. Clicking this link should take the user back to the `/docs/reference` page

### CODE REVIEW

**Comment by Isaiah-Hamilton** [comment]
Closes https://github.com/supabase/supabase/issues/6152 because the table will no longer be there.

**Comment by Isaiah-Hamilton** [comment]
cc @kiwicopple all updated

image

https://user-images.githubusercontent.com/70828596/172093169-1185aec3-aed7-4a0d-aa73-6f57e7414c83.mp4

File: web/docs/reference.mdx [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+
+## Client Libraries
+
+<div class="container" style={{ padding: 0 }}>
+  <div class="row is-multiline">
+    {/* Database */}
+    <div class="col col--6">
+      <Link class="card" to="/docs/reference/javascript/installing" style={{ height: '100%' }}>
+        <div class="card__body">
+          <h4>Javascript Client Library</h4>
```
Comment (Isaiah-Hamilton) [inline web/docs/reference.mdx:45]: I think its a little redundant to restate that Javascript is a client library, and if we want to restate that its a client library then we should do the same for Dart.

Comment (kiwicopple) [inline web/docs/reference.mdx:45]:
```suggestion
          <h4>JavaScript</h4>
```

Comment (kiwicopple) [reply web/docs/reference.mdx]: good call!

**Review by Isaiah-Hamilton — APPROVED** [review]

**Pull request merged**